sdks/java/extensions/arrow/src/test/java/org/apache/beam/sdk/extensions/arrow/ArrowConversionTest.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.arrow.vector.TimeStampMicroTZVector;
+import org.apache.arrow.vector.TimeStampMilliTZVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.complex.ListVector;
+import org.apache.arrow.vector.types.FloatingPointPrecision;
+import org.apache.arrow.vector.types.TimeUnit;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.util.Text;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.Schema.Field;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.hamcrest.collection.IsIterableContainingInOrder;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ArrowConversionTest {
+
+  private BufferAllocator allocator;
+
+  @Before
+  public void init() {
+    allocator = new RootAllocator(Long.MAX_VALUE);
+  }
+
+  @After
+  public void teardown() {
+    allocator.close();
+  }
+
+  @Test
+  public void toBeamSchema_convertsSimpleArrowSchema() {
+    Schema expected =
+        Schema.of(Field.of("int8", FieldType.BYTE), Field.of("int16", FieldType.INT16));
+
+    org.apache.arrow.vector.types.pojo.Schema arrowSchema =
+        new org.apache.arrow.vector.types.pojo.Schema(
+            ImmutableList.of(
+                field("int8", new ArrowType.Int(8, true)),
+                field("int16", new ArrowType.Int(16, true))));
+
+    assertThat(ArrowConversion.toBeamSchema(arrowSchema), equalTo(expected));
+  }
+
+  @Test
+  public void rowIterator() {
+    org.apache.arrow.vector.types.pojo.Schema schema =
+        new org.apache.arrow.vector.types.pojo.Schema(
+            asList(
+                field("int32", new ArrowType.Int(32, true)),
+                field("float64", new ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE)),
+                field("string", new ArrowType.Utf8()),
+                field("timestampMicroUTC", new ArrowType.Timestamp(TimeUnit.MICROSECOND, "UTC")),
+                field("timestampMilliUTC", new ArrowType.Timestamp(TimeUnit.MILLISECOND, "UTC")),
+                field(
+                    "int32_list",
+                    new ArrowType.List(),
+                    field("int32s", new ArrowType.Int(32, true))),
+                field("boolean", new ArrowType.Bool()),
+                field("fixed_size_binary", new ArrowType.FixedSizeBinary(3))));
+
+    Schema beamSchema = ArrowConversion.toBeamSchema(schema);
+
+    VectorSchemaRoot expectedSchemaRoot = VectorSchemaRoot.create(schema, allocator);
+    expectedSchemaRoot.setRowCount(16);
+    for (FieldVector vector : expectedSchemaRoot.getFieldVectors()) {
```

Comment:
I believe VectorSchemaRoot has an allocateNew

Reply:
[You're right](https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/VectorSchemaRoot.html#allocateNew--), let's use it here.